open-company/open-company-web
# REFACTOR: Move all section related code to action creators and store.

Actions involved:

```
:boards-load-other
:board-get
:board-seen
:board-nav-away
:board
:board-delete
:whats-new/finish
:section-edit-save
:section-edit-save/finish
:section-edit/dismiss
:private-section-user-add
:private-section-user-remove
:private-section-kick-out-self
:private-section-kick-out-self/finish
:container/section-change
```

To test:

`:boards-load-other`
- Login and go to all posts.
- [x] In the console do you see requests for the existing sections? `/boards/<board-name>` followed by `Dispatching action: :section`

`:board-get`  This also tests `:board` (the old way of handling the API return)
Technically was tested above but you can also
- Login and click on a section.
- [x] The posts for that section load and you will see the same `Dispatching action :section` message.
Draft test
- Create a new draft
- [x] Does the draft section load with the draft post?

`:board-seen`
Repeating the same steps for `:board-get` 
- [x] Do you see `Dispatching action :section-seen` in the console?

`:board-nav-away`
- Switch boards from the `:board-get` test. You have to be in the old "entries" view .
- [x] Do you see `Dispatching action :section-nav-away`?

`:board-delete`
- [x] Is the section deleted when clicking on the section editor and clicking delete?

```
:section-edit-save
:section-edit-save/finish
:section-edit/dismiss
```
- [x] Does create a new section work?
- [x]  Does editing the section name work?

```
:private-section-user-add
:private-section-user-remove
:private-section-kick-out-self
:private-section-kick-out-self/finish
```
- [x] Make a board private and add yourself. Does saving this work?
- [x]  Make a board private and add someone else.
- [x] Remove the other user and save.
- [x] Remove yourself and save.

`:container/section-change`
This was added to handle websocket messages from the change service.
- Login as user A and user B
- As user A click on a section.
- [x] As user B add a post to the same section and does it show up for user A?
